proto/zitadel/user/v3alpha/authenticator.proto

```diff
@@ -0,0 +1,468 @@
+  string username_id = 1;
+  // The user's unique username. It is used for identification during authentication.
+  string username = 2 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"gigi-giraffe\"";
+    }
+  ];
+  // By default usernames must be unique across all organizations in an instance.
+  // This option allow to restrict the uniqueness to the user's own organization.
+  // As a result, this username can only be used if the authentication is limited
+  // to the corresponding organization.
+  //
+  // This can be useful if you provide multiple usernames for a single user, where one
+  // if specific to your organization, e.g.:
+  // - gigi-giraffe@zitadel.com (unique across organizations)
+  // - gigi-giraffe (unique only inside the ZITADEL organization)
+  bool is_organization_specific = 3;
+}
+
+message SetUsername {
+  // Set the user's username. This will be used for identification during authentication.
+  string username = 1 [
+    (validate.rules).string = {min_len: 1, max_len: 200},
+    (google.api.field_behavior) = REQUIRED,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      min_length: 1,
+      max_length: 200,
+      example: "\"gigi-giraffe\"";
+    }
+  ];
+  // By default username must be unique across all organizations in an instance.
+  // This option allow to restrict the uniqueness to the user's own organization.
+  // As a result, this username can only be used if the authentication is limited
+  // to the corresponding organization.
+  //
+  // This can be useful if you provide multiple usernames for a single user, where one
+  // if specific to your organization, e.g.:
+  // - gigi-giraffe@zitadel.com (unique across organizations)
+  // - gigi-giraffe (unique only inside the ZITADEL organization)
+  bool is_organization_specific = 2;
+}
+
+message Password {
+  // States the time the password was last changed.
+  //(--TODO:do we want to include that in the list or is there a separate call?--)
+  google.protobuf.Timestamp last_changed = 1 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"2019-04-01T08:45:00.000000Z\"";
+    }
+  ];
+}
+
+message WebAuthN {
+  // unique identifier of the WebAuthN authenticator.
+  string web_auth_n_id = 1 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"69629023906488334\""
+    }
+  ];
+  // Name of the WebAuthN authenticator. This is used for easier identification.
+  string name = 2 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"fido key\""
+    }
+  ];
+  // State whether the WebAuthN registration has been completed.
+  //(--TODO:only verified or return verification state?--)
+  bool is_verified = 3;
+  //(--TODO:do we need to know if the user has been verified in the initial registration or do we only care on authentication?--)
+  //(--bool user_verified = 4;--)
```

Comment:
do we need to know if the user has been verified in the initial registration or do we only care on authentication?

Reply:
I would handle it like in the session API, you can still create a user but just only use it to authentication if verified.

Reply:
this boolean is about the WebAuthN user_verification, which mean the user was verified by a PIN or other mechanism, which makes it a MFA in one step. Without that only the private key is checked (2FA).
in the session API we already handle it that way in the authentication, but we there have two different preregistered webauthn device pools, one with user verification and one without.
These two pools create problems, since a device typically will only allow to be registered once, forgetting about the first registration after the second. We therefore want to combine them.
But while writing this i noticed, that it's probably the best to return this information to the user as well, as we need to store it to be able to return the `allowed credentials` in the authentication flow.

Reply:
True, I'm with you on that